react-graph/src/Plot.tsx

```diff
@@ -536,14 +539,56 @@ const Plot: React.FunctionComponent<IProps> = (props) => {
-      else setSelectedMode(s as ('zoom'|'pan'|'select'))
+      else if (s === "capture") {
+        setPhotoReady(true);
+        if (props.onCapture !== undefined) heightChange.current.captureID = props.onCapture(heightChange.current.extraNeeded);
+      }
+      else setSelectedMode(s as ('zoom-rectangular' | 'zoom-vertical' | 'zoom-horizontal' | 'pan' | 'select'))
     }, [tDomain, Reset, props.onDataInspect]);
 
+    /* const getContrainedTDomain = React.useCallback((newYDomain: [number,number][]): [number,number] => {
+      const tMinArray: number[] = [];
+      const tMaxArray: number[] = [];
+
+      [...data.values()].forEach((series: IDataSeries) => {
+        // ToDo: Get rid of getData, ask each bucket if our time domain about their min/max instead to discard buckets then look for the bucket where a value we can see falls in first and last
+        const data = series.getData([Number.MIN_SAFE_INTEGER, Number.MAX_SAFE_INTEGER], false);
+        const axis = AxisMap.get(series.axis);
+        const valMin = data.find(val => val[1] >= newYDomain[axis][0] && val[1] <= newYDomain[axis][1]);
+
+        if (valMin !== undefined) tMinArray.push(valMin[0]);
+        const valMaxInd = findLastIndex(data, val => val[1] >= newYDomain[axis][0] && val[1] <= newYDomain[axis][1]);
+        if (valMaxInd !== -1) tMaxArray.push(data[valMaxInd][0]);
+      })
+
+      const tMin = Math.min(...tMinArray);
+      const tMax = Math.max(...tMaxArray);
+      if (!isNaN(tMin) && !isNaN(tMax) && isFinite(tMin) && isFinite(tMax)) return [tMin, tMax];
+      else return tDomain;
+    }, [data, tDomain]); */
+
+    const getConstrainedYDomain = React.useCallback((newTDomain: [number, number]): [number,number][] => {
+      const dataReducerFunc = (result: number[], series: IDataSeries, func: (tDomain: [number, number]) => number|undefined, axis: number) => {
+        // This part of the data may not belong to the axis we care about at the moment
+        const dataAxis = AxisMap.get(series.axis);
+        if (axis === dataAxis) {
+          const value =  func(newTDomain);
+          if (value !== undefined) result.push(value);
+        }
+        return result;
+      }
+      return yDomain.map((oldDomain, axis) => {
+        const yMin = Math.min(...[...data.values()].reduce((result: number[], series: IDataSeries) => dataReducerFunc(result, series, series.getMin, axis), []));
+        const yMax = Math.max(...[...data.values()].reduce((result: number[], series: IDataSeries) => dataReducerFunc(result, series, series.getMax, axis), []));
+        if (!isNaN(yMin) && !isNaN(yMax) && isFinite(yMin) && isFinite(yMax)) return [yMin, yMax];
+        return yDomain[axis];
+      });
+    }, [data, yDomain]);
+
     function handleMouseWheel(evt: any) {
           if (props.zoom !== undefined && !props.zoom)
               return;
-          if (!(selectedMode === 'zoom'))
+          if (!(selectedMode.match(/(?:zoom)/i) != null))
```

Comment:
May be easier to read 
```suggestion
          if (!selectedMode.contains('zoom'))
```